### PR TITLE
Add release guide

### LINKED ISSUE #47: Document release/versioning process
Releasing new versions of testcontainers and all it's little subcrates can be quite difficult. In particular, figuring out what the next version should be is not trivial.

We should create a guide that documents how versioning works within the `testcontainers-rs` ecosystem.

### PR DESCRIPTION
**Pull request by thomaseizinger**
Resolves #47.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,11 +19,18 @@ The repository is structured into the several crates.
 
 - `core`: Contains the core traits and structs necessary for the testcontainers ecosystem.
 - `cli_client`: Contains an implementation of the `Docker` trait that uses the `docker`-CLI to issue commands.
-- The folder `images` contains several crates named after the respective docker image. Each crate adds support for one particular image. This allows to selectively import the images you need.
+- The folder `images` contains several crates named after the respective docker image. 
+Each crate adds support for one particular image. This allows to selectively import the images you need.
```
Comment (D4nte) [inline README.md]: Randomly add a new line?

File: README.md [hunk 2]
```diff
@@ -19,11 +19,18 @@ The repository is structured into the several crates.
...
 
 Last but not least:
 `testcontainers`: This is a meta crate that bundles all these crates together for convenient usage.
 
+## Versioning
+
+To not make it too inconvenient, the meta crate `testcontainers` cannot strictly follow semver.
+Thus you are encouraged to at least depend on the minor version of the library, i.e. `X.Y`. 
+If you are interested in the versioning approach, chosen by this crate, checkout the [release guide](./RELEASING.md).
```
Comment (D4nte) [inline README.md]:
```suggestion
If you are interested in this crate versioning approach, checkout the [release guide](./RELEASING.md).
```

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+## Core
+
+The `tc_core` crate provides the main types like the `Image` trait or the `Container` struct.
+All other crates depend on this. 
+Any backwards-incompatible change must be appropriately reflected through versioning. 
+Very likely, all downstream crates like concrete images and clients will require an equivalent version update.
+
+## Images
+
+Versioning the image crates can be quite tricky and the consequences are not super obvious.
```
Comment (D4nte) [inline RELEASING.md]:
```suggestion
Versioning the image crates can be quite tricky and the consequences are not obvious. 
```

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+When releasing a new version of an image crate, two aspects should be considered when choosing the next version:
+
+- Compatibility of the Rust API
+- Compatibility of the exposed Docker image
+
+In regards to the Rust API of an Image (the `XYArgs` struct or the `XY` Image struct itself), the usual Rust rules apply. 
+For example, removing or renaming a previously public field is a breaking change.
+
+Changes to the used Docker image a bit trickier. 
+It very much depends on the what changed in the underlying Docker image.
```
Comment (D4nte) [inline RELEASING.md]:
```suggestion
It very much depends on what changed in the underlying Docker image. 
```

File: RELEASING.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+Changing the docker-tag that the `tc_coblox_bitcoincore` crate is referring to is thus a breaking change for this crate and should be appropriately reflected in the version number.  
+
+## The meta crate
+
+The meta-crate plays an interesting role in this ecosystem. 
+Its main purpose is convenience by re-exporting a **valid** and **working** set of images and clients.
+
+To achieve this, the version of `tc_core` needs to be compatible among all the dependencies (i.e. the image and client crates). 
+
+The meta-crate doesn't have any code on it's own, so what does the version number actually represent?
```
Comment (D4nte) [inline RELEASING.md]:
```suggestion
The meta-crate doesn't have any code on its own, so you may wonder what does the version number actually represent.
```

File: RELEASING.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+It represents, which images you can use, i.e. a set of features.
```
Comment (D4nte) [inline RELEASING.md]:
```suggestion
It represents which images you can use, i.e. a set of features.
```

File: RELEASING.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+Because it is a meta-crate, reflecting all major version upgrades as a major version upgrade would result in an explosion of versions.
+
+For this reason, breaking changes in downstream system **will not** be propagated as such to users of the meta crate.
+What is meant by that?
```
Comment (D4nte) [inline RELEASING.md]:
```suggestion
```

File: RELEASING.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+This means, a version upgrade vom `tc_coblox_bitcoincore` from `1.0` to `2.0` **does not** result in a major version upgrade of the `testcontainers` meta crate.
```
Comment (D4nte) [inline RELEASING.md]:
```suggestion
This means a version upgrade of `tc_coblox_bitcoincore` from `1.0` to `2.0` **does not** result in a major version upgrade of the `testcontainers` meta crate.
```

File: RELEASING.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+Instead, `testcontainers` will only have its minor version bumped.
+Because of that, users of `testcontainers` are encouraged to at least specify the minor version when depending on `testcontainers`.
+
+## What about the problems this causes?
+
+Are are some potential problems with this approach. Remember, the meta-crate only exists as a **convenience** for the majority of users.
```
Comment (D4nte) [inline RELEASING.md]:
```suggestion
There are some potential problems with this approach.
Remember, the meta-crate only exists as a **convenience** for the majority of users.
```

File: README.md (hunk 1 above)
Comment (thomaseizinger) [reply README.md]: Wanted to follow the pattern of one sentence per line.

Comment (D4nte) [reply README.md]: But you still have 2 sentences on the line.

Comment (thomaseizinger) [reply README.md]: Good point :D 
Will change that.

**Pull request merged**